src/Generator/Writer.cs

```diff
@@ -50,67 +51,104 @@ public static void Write(OutputData data)
 
-                        foreach (var function in group.NativeFunctions)
-                        {
-                            bool postfixName = group.NativeFunctionsWithPostfix.Contains(function);
-                            documentation.TryGetValue(function, out FunctionDocumentation? functionDocumentation);
-                            WriteNativeMethod(writer, function, postfixName, functionDocumentation);
-                        }
+        private static void WriteNativeMethod(IndentedTextWriter writer, NativeFunction function, bool postfixName, FunctionDocumentation? documentation)
+        {
+            // Write delegate field initialized to the lazy loader.
+            // Write public function definition that calls delegate.
+            // Write lazy loader function.
+            string name, returnType;
+            StringBuilder paramNames, delegateTypes, signature;
+            GetNativeFunctionSignature(function, postfixName, SwapUnderlyingTypeForPrimitive, out name, out paramNames, out delegateTypes, out signature, out bool _, out returnType);
 
-                        scope?.Dispose();
-                    }
+            writer.WriteLine($"internal static delegate* unmanaged<{delegateTypes}> _{name}_fnptr = &{name}_Lazy;");
+
+            writer.WriteLine($"[UnmanagedCallersOnly]");
+            writer.WriteLine($"private static {returnType} {name}_Lazy({signature})");
+            using (writer.CsScope())
+            {
+                // Dotnet gurantees you can't get torn values when assigning functionpointers, assuming proper allignment which is default.
+                writer.WriteLine($"_{name}_fnptr = (delegate* unmanaged<{delegateTypes}>){LoaderBindingsContext}.GetProcAddress(\"{function.EntryPoint}\");");
+
+                if (function.ReturnType is not CSVoid)
+                {
+                    writer.WriteLine($"return _{name}_fnptr({paramNames});");
+                }
+                else
+                {
+                    writer.WriteLine($"_{name}_fnptr({paramNames});");
                 }
             }
 
-            writer.Flush();
+            writer.WriteLine();
         }
 
-        private static void WriteNativeMethod(IndentedTextWriter writer, NativeFunction function, bool postfixName, FunctionDocumentation? documentation)
+        private static string SwapUnderlyingTypeForPrimitive(BaseCSType type)
+        {
+            if (type is CSStruct csstruct)
+            {
+                return csstruct.UnderlyingType?.ToCSString() ?? throw new Exception("A struct didnt contain an underlying type.");
+            }
+            else if (type is CSEnum csenum)
+            {
+                return "uint";
+            }
+            else if (type is CSPointer cspointer)
+            {
+                return SwapUnderlyingTypeForPrimitive(cspointer.BaseType) + "*";
+            }
+            else
+            {
+                return type.ToCSString();
+            }
+        }
```

Comment:
This function doesn't need to be recursive, it would probably make it simpler if it was using just a while-loop.